Lib/_pyrepl/commands.py

```diff
@@ -59,6 +61,8 @@ def do(self) -> None:
 
 
 class KillCommand(Command):
+    modifies_buffer = True
```

Comment:
I'm not fond of having commands implementation depending on VI mode. Insteaad, I would suggest having two sets of commands that are distinct classes instead, e.g.:

``` 
class right(Command): ...
class vi_right(right): ...
```

where you can override whatever you need. Having each command being aware of the VI mode makes the implementation too rigid.

When entering a specific vi mode, the command handler should be replaced by the corresponding vi-compatible handler.